src/verifier/verifier.rs

```diff
@@ -19,7 +27,535 @@ pub fn build_symbol_table(ast: &mut Whamm, err: &mut ErrorGen) -> SymbolTable {
+                                    &rhs_loc.line_col,
+                                    None,
+                                );
+                                self.err.type_check_error(
+                                    false,
+                                    format! {"Type Mismatch, lhs:{:?}, rhs:{:?}", lhs_ty, rhs_ty},
+                                    &Some(loc.line_col),
+                                );
+
+                                None
+                            }
+                        }
+                        BinOp::GT | BinOp::LT | BinOp::GE | BinOp::LE => {
+                            if lhs_ty == DataType::I32 && rhs_ty == DataType::I32 {
+                                Some(DataType::Boolean)
+                            } else {
+                                // using a struct in parser to merge two locations
+                                let loc = crate::parser::types::Location::from(
+                                    &lhs_loc.line_col,
+                                    &rhs_loc.line_col,
+                                    None,
+                                );
+                                self.err.type_check_error(
+                                    false,
+                                    format! {"Type Mismatch, lhs:{:?}, rhs:{:?}", lhs_ty, rhs_ty},
+                                    &Some(loc.line_col),
+                                );
+
+                                None
+                            }
+                        }
+                    }
+                } else {
+                    let loc = crate::parser::types::Location::from(
+                        &lhs_loc.line_col,
+                        &rhs_loc.line_col,
+                        None,
+                    );
+                    self.err.type_check_error(
+                        false,
+                        "Can't get type of lhs or rhs of this binary operation".to_string(),
+                        &Some(loc.line_col),
+                    );
+                    None
+                }
+            }
+            #[allow(unused_variables)]
+            Expr::VarId {
+                name,
+                loc,
+                is_comp_provided,
+            } => {
+                // get type from symbol table
+                // println!("curr scope: {:?}", self.table.get_curr_scope());
+                // let _ = self.table.enter_scope(); // Alex: adding enter_scope here doens't seems to have an effect here (which is reasonable)
+                // println!("curr scope: {:?}", self.table.get_curr_scope());
+
+                if let Some(id) = self.table.lookup(name) {
+                    // println!("LOOK AT ME Var id: {:?}", self.table.get_record(id));
+                    if let Some(rec) = self.table.get_record(id) {
+                        if let Record::Var { ty, .. } = rec {
+                            return Some(ty.to_owned());
```

Comment:
This is moving the type of the value out of the symbol table, instead this should be a clone of the type since we'd be using this later.